features/layer_lock.h

```diff
@@ -82,7 +82,22 @@ void layer_lock_invert(uint8_t layer);
 //   }
 void layer_lock_set_user(layer_state_t locked_layers);
 
+#if LAYER_LOCK_IDLE_TIMEOUT > 0
+
+//If you want to automatically disable layer lock if no keys are pressed after a
+//configurable timeout, define LAYER_LOCK_IDLE_TIMEOUT to a value between 100
+// and 30000 (in milliseconds) in your config.h file and add the following
+//functions as described below.
+
+//Add this function below your process_record_user function, preferrably right below
```

Comment:
Aha, so `process_layer_lock_timer()` is being called separately from `process_record_user()`. Would there be any downside to restarting the timer directly in `process_layer_lock()`? That seems good so that the user has one less thing to do to use the library.

Something like:

```c
bool process_layer_lock(uint16_t keycode, keyrecord_t* record,
                        uint16_t lock_keycode) {
#if LAYER_LOCK_IDLE_TIMEOUT > 0
  if (locked_layers) {  // If any layers are locked, restart the timer.
    layer_lock_timer = record->event.time;
  }
#endif  // LAYER_LOCK_IDLE_TIMEOUT > 0

  // Rest of function as before...
```

Reply:
I don't really see any reason there should be a problem doing that. I'll try it tonight and make sure it doesn't break anything, but I don't expect it will.

Reply:
That's mysterious. I remember seeing a note somewhere (maybe in the auto shift implementation) claiming that `record->event.time` isn't always reliable, yet plenty of QMK's core features make use of `record->event.time`, so I didn't know what to make of this. It seems worth to try whether it helps to change the update line to

```c
layer_lock_timer = timer_read();
```

Reply:
I did actually try with the timer_read() function as well and got the same result. I really don't understand why this particular problem crops up when updating the timer inside process_layer_lock, it doesn't make a lot of sense to me just yet. I also had debug information being printed and the timer was updated as expected on each keypress, and if I only pressed a few keys and stopped with the layer lock still active it would run out the full timer from the last keypress, so it does seem to update correctly. Very strange.

Reply:
Moving the timer_read to the top of the process_layer_lock function on all keypresses fixed this issue. Only the timer read task now has to be added to matrix_scan by the user and define the LAYER_LOCK_IDLE_TIMEOUT in config.h.